### PR TITLE
Sub story1 3

### PR DESCRIPTION
**Pull request by NeuroticWriter**
SplashScreen made with animation and timer.

### CODE REVIEW

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by DevilDewzone on 7/5/2016.
+ */
+public class SplashScreenActivity extends AppCompatActivity {
+
+    Handler handler;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter make these private

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by DevilDewzone on 7/5/2016.
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java:19]: @NeuroticWriter summarize what the class does

File: app/src/main/res/layout/splash_screen_layout.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="fill_parent"
+    android:layout_height="match_parent"
+    android:background="#f6fffd"
+    android:paddingBottom="0dp"
```
Comment (manas-raj-shrestha) [inline app/src/main/res/layout/splash_screen_layout.xml]: @NeuroticWriter why is padding defined here if there no padding?

File: app/src/main/res/layout/splash_screen_layout.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="fill_parent"
```
Comment (manas-raj-shrestha) [inline app/src/main/res/layout/splash_screen_layout.xml]: @NeuroticWriter fill_parent is deprecated

File: app/src/main/res/layout/splash_screen_layout.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    android:paddingTop="0dp"
+    tools:context="com.lftechnology.S">
+
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:src="@drawable/applogo"
+        android:id="@+id/splashscreen_img"
+        android:layout_centerVertical="true"
+        android:layout_centerHorizontal="true" />
```
Comment (manas-raj-shrestha) [inline app/src/main/res/layout/splash_screen_layout.xml]: @NeuroticWriter isn't centerInParent same as using centerVertical and centerHorizontal?

File: app/src/main/res/layout/splash_screen_layout.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        android:id="@+id/third_line"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="49dp"
+        android:text="Please wait for a moment while we prepare for you."
+        android:textColor="#8f0634" />
+
+
```
Comment (manas-raj-shrestha) [inline app/src/main/res/layout/splash_screen_layout.xml]: @NeuroticWriter remove the unnecessary empty line

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                 public void run() {
+                     gotoNext();
+
+                 }
+             },
+             3000
+     );
+    }
+
+    public void gotoNext(){
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter why is this method public? i don't think this method is used by any other classes.

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+//        //Third Line animation fade in and out
+//        TextView txtThird = (TextView) findViewById(R.id.third_line);
+//        Animation ThirdIn = AnimationUtils.loadAnimation(this, R.anim.fade_in_animation);
+//        txtThird.setAnimation(ThirdIn);
+//        Animation thirdOut= AnimationUtils.loadAnimation(this, R.anim.fade_out_animation);
+//        txtThird.setAnimation(thirdOut);
+//
+        //MainActivty passing after splash
+     // SPlash to next activity wait timer
+     new Timer().schedule(
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter look into handler.postdelayed

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 1 above)
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter is handler even used in this class? if no then why is it here?

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by DevilDewzone on 7/5/2016.
+ */
+public class SplashScreenActivity extends AppCompatActivity {
+
+    Handler handler;
+    Runnable delayRun;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter delayRun is never used.

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+        /* previous code for full screen
+            requestWindowFeature(Window.FEATURE_NO_TITLE);
+            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        */
+        setContentView(R.layout.splash_screen_layout);
+
+
+        //For fade in out animation
+        //Setting the alpha values
+        AlphaAnimation fadeIn = new AlphaAnimation(0.0f, 1.0f);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java:56]: @NeuroticWriter better use onCreate just for initializing views. Make another method call setAnimations and set the animations to the view from that method.

File: app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -7,14 +7,20 @@
...
+            android:label="@string/app_name">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
-
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+
         </activity>
+        <activity android:name=".MainActivity">
+            android:theme=@style/Theme.AppCompat.NoActionBar"
```
Comment (manas-raj-shrestha) [inline app/src/main/AndroidManifest.xml]: @NeuroticWriter what is this? does this not give an error?

File: app/src/main/java/com/lftechnology/activitylogger/MainActivity.java [hunk 12]
```diff
@@ -1,13 +1,23 @@
...
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
 
 public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        //  View decorView = getWindow().getDecorView();
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/MainActivity.java]: @NeuroticWriter remove the unnecessary comments

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+        */
+
+        //Start and end for the text animation
+        txtFirst.startAnimation(fadeIn);
+        txtFirst.startAnimation(fadeOut);
+        txtSecond.setAnimation(fadeIn);
+        txtSecond.setAnimation(fadeOut);
+        txtThird.setAnimation(fadeIn);
+        txtThird.setAnimation(fadeOut);
+        fadeIn.setDuration(1500);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: @NeuroticWriter for constants, like 1500, create a private static final int ANIMATION_DURATION variable and use it.

Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: Done

File: app/src/main/java/com/lftechnology/activitylogger/MainActivity.java (hunk 12 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/MainActivity.java]: done

File: app/src/main/AndroidManifest.xml (hunk 11 above)
Comment (NeuroticWriter) [inline app/src/main/AndroidManifest.xml]: It removes the action bar in the mainactivity.
removed it fixed it.
no it doesn't gave an error while running with it

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 10 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java:56]: done

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 9 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: removed

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 7 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: done

File: app/src/main/res/layout/splash_screen_layout.xml (hunk 6 above)
Comment (NeuroticWriter) [inline app/src/main/res/layout/splash_screen_layout.xml]: done

File: app/src/main/res/layout/splash_screen_layout.xml (hunk 5 above)
Comment (NeuroticWriter) [inline app/src/main/res/layout/splash_screen_layout.xml]: tried and replaced.
didn't knew before.

File: app/src/main/res/layout/splash_screen_layout.xml (hunk 4 above)
Comment (NeuroticWriter) [inline app/src/main/res/layout/splash_screen_layout.xml]: done replaced with match parent

File: app/src/main/res/layout/splash_screen_layout.xml (hunk 3 above)
Comment (NeuroticWriter) [inline app/src/main/res/layout/splash_screen_layout.xml]: removed unused lines

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 2 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java:19]: done

File: app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java (hunk 8 above)
Comment (NeuroticWriter) [inline app/src/main/java/com/lftechnology/activitylogger/SplashScreenActivity.java]: done

**Pull request merged**